packages/babel-plugin-transform-es2015-parameters/src/rest.js

```diff
@@ -177,14 +178,30 @@ function optimiseIndexGetter(path, argsId, offset) {
     scope.push({ id: temp, kind: "var" });
     path.parentPath.replaceWith(restIndexImpure({
       ARGUMENTS: argsId,
+      OFFSET: offsetLiteral,
       INDEX: index,
       REF: temp,
     }));
   } else {
-    path.parentPath.replaceWith(restIndex({
+    const parentPath = path.parentPath;
+    parentPath.replaceWith(restIndex({
       ARGUMENTS: argsId,
+      OFFSET: offsetLiteral,
       INDEX: index,
     }));
+
+    // See if we can statically evaluate the first test (i.e. index < offset)
+    // and optimize the AST accordingly.
+    const offsetTestPath = parentPath.get("test").get("left");
+    const valRes = offsetTestPath.evaluate();
+    if (valRes.confident) {
+      if (valRes.value === true) {
+        parentPath.replaceWith(parentPath.scope.buildUndefinedNode());
+      }
+      else {
+        parentPath.get("test").replaceWith(parentPath.get("test").get("right"));
```

Comment:
Also, replaceWith accepts path: https://github.com/babel/babel/blob/1e55653ac173c3a59f403a7b5269cf6d143d32d9/packages/babel-traverse/src/path/replacement.js#L106.

But I can change the line to `parentPath.get("test").replaceWith(parentPath.get("test").get("right").node)` if you think it's better.